src/libmongoc/src/mongoc/mongoc-scram-private.h

```diff
@@ -123,6 +123,346 @@ _mongoc_sasl_prep_required (const char *str);
 char *
 _mongoc_sasl_prep (const char *in_utf8, int in_utf8_len, bson_error_t *err);
 
+/* returns how many bytes a UTF-8 character is. */
+int
+_mongoc_utf8_char_length (const char *c);
+
+/* returns how many characters are in a UTF-8 string. Returns -1 on error. */
+int
+_mongoc_utf8_string_length (const char *s);
+
+/* returns whether a UTF-8 character is valid or not. */
+bool
+_mongoc_utf8_is_valid (const char *c, int length);
+
+/* returns whether a character is between two limits (inclusive). */
+bool
+_mongoc_char_between_chars (const char c, const char lower, const char upper);
+
+/* returns whether a codepoint exists in the specified table. The table format
+ * is that the 2*n element is the lower bound and the 2*n + 1 is the upper bound
+ * (both inclusive). */
+bool
+_mongoc_is_code_in_table (unsigned int code,
+                          const unsigned int *table,
+                          int size);
+
+/* converts a UTF-8 character to unicode code point. */
+unsigned int
+_mongoc_utf8_to_unicode (const char *c, int length);
```

Comment:
Suggest renaming function to `_mongoc_utf8_get_first_codepoint` and updating the description accordingly.